pkg/utils/gardener/operator/operator.go

```diff
@@ -0,0 +1,38 @@
+// Copyright 2023 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package operator
+
+import (
+	"context"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	operatorv1alpha1 "github.com/gardener/gardener/pkg/apis/operator/v1alpha1"
+	operatorclient "github.com/gardener/gardener/pkg/operator/client"
+	kubernetesutils "github.com/gardener/gardener/pkg/utils/kubernetes"
+)
+
+// SeedIsGarden returns 'true' if the cluster is registered as a Garden cluster.
+func SeedIsGarden(ctx context.Context, seedClient client.Reader) (bool, error) {
```

Comment:
Aren't those rather "gardenletutils"?

Btw, why is this suddenly necessary? 👀

Reply:
> Aren't those rather "gardenletutils"?

I can rename the package, no strong opinion.

> Btw, why is this suddenly necessary? 👀

`pkg/extensions` imports `pkg/utils/gardener` and since we now use `operatorclient.RuntimeScheme` in `SeedIsGarden`, `pkg/extensions` has a transitive dependency to `pkg/operator/client`. This is not allowed per import-restriction which probably makes sense 😄

Reply:
Okay 🫣😄